tensorflow/python/kernel_tests/array_ops_test.py

```diff
@@ -523,7 +523,15 @@ def eval_if_tensor(x):
       np_spec = eval_if_tensor(spec)
       self.test.assertAllEqual(self.x_np[np_spec], tensor)
       return tensor
-
+    
+    if (isinstance(spec, np.ndarray) and len(spec.shape) > 1) or \
+    (isinstance(spec, ops.Tensor) and (spec.shape.dims is not None) and 
+    len(spec.shape) > 1):
```

Comment:
@alextp, When the slice_spec is `tensor` or `np.array`, and the tensor is multi-dimension, we should deal with it separately here.

Reply:
Probably, yes.

You might have to push the logic of this change inside the C++ code for StridedSlice :-/

Reply:
@alextp, Sorry, I have some confusion here, as my understanding, for the condition above, we will go through the logic `gather_nd`, not `strided_slice`. Could you please give me some more reference? Thanks.

Reply:
What I was alluding to is that if your input is a tensor of unknown shape (which can happen in tensorflow) it'll miss this static python-time check.

So we need to make this static check a runtime check, which means it has to happen inside some C++ op.

Reply:
@alextp, Sorry for late response, and Thanks for kind explanation. 
I could understand what you say now, but as my understanding, for NumPy, the spec shape must be specified when we use the slice operation.
And I add the logic above just because the logic below
```
np_specs = []
np_specs.append(eval_if_tensor(s))
```
for np_specs which is like `[np.array([[0, 1, 1], [1, 0, 1], [1, 0, 2]])], we could not correctly evaluate `self.x_np[tuple(np_specs)]`. 
So any suggestions here?

Reply:
In numpy all shapes are always specified because numpy does not distinguish runtime from graph building time, but tensorflow does.

So we still need to handle tensors of unknown shapes as if we knew their shapes, which means doing things at run time.